MekHQ/unittests/mekhq/EventSpy.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2020 - The MegaMek Team. All rights reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+import megamek.common.annotations.Nullable;
+import megamek.common.event.MMEvent;
+import megamek.common.event.Subscribe;
+import mekhq.campaign.event.*;
+
+/**
+ * Provides a list of events captured during its lifetime.
+ * Use this as part of a try-with-resources block.
+ *
+ * If you need to listen to a new event, add a handler to this
+ * class.
+ */
+public class EventSpy implements AutoCloseable {
+    private final List<MMEvent> events = new ArrayList<>();
+
+    /**
+     * Creates a new EventSpy and registers it with
+     * MekHQ's event bus.
+     */
+    public EventSpy() {
+        MekHQ.registerHandler(this);
+    }
+
+    /**
+     * Deregisters this instance from MekHQ's event bus.
+     */
+    public void close() {
+        MekHQ.unregisterHandler(this);
+    }
+
+    /**
+     * Gets the list of events which occurred.
+     * @return The list of events which occurred, in order
+     *         received.
```

Comment:
This is a perfect example of an accessibility issue where the comment is split over two lines. One line would be best, even if it goes slightly over the expected length, when only a single word is split to make it . This is irregardless of screen size, because this actually leads to far longer comprehension.